website/docs/sdk-reference/rust.mdx

```diff
@@ -0,0 +1,705 @@
+    ("doubleSetting", Value::Float(1.2)),
+    ("stringSetting", Value::String("test".to_owned())),
+].into();
+
+let client = Client::builder("localhost")
+    // highlight-next-line
+    .overrides(Box::new(map), OverrideBehavior::LocalOnly)
+    .build()
+    .unwrap();
+```
+
+## `get_all_keys()`
+
+You can query the keys of each feature flag and setting with the `get_all_keys()` method.
+
+```rust
+let client = Client::new("#YOUR-SDK-KEY#").unwrap();
+let keys = client.get_all_keys().await;
+```
+
+## `get_all_values()`
+
+Evaluates and returns the values of all feature flags and settings. Passing a [User Object](#user-object) is optional.
+
+```rust
+let client = Client::new("#YOUR-SDK-KEY#").unwrap();
+let values = await client.get_all_values(None).await;
+
+// invoke with User Object
+let user = User::new("#UNIQUE-USER-IDENTIFIER#");
+let values_with_user = await client.get_all_values(Some(user)).await;
+```
+
+## `get_all_details()`
+
+Evaluates and returns the values along with evaluation details of all feature flags and settings. Passing a [User Object](#user-object) is optional.
+
+```rust
+let client = Client::new("#YOUR-SDK-KEY#").unwrap();
+let details = await client.get_all_details(None).await;
+
+// invoke with User Object
+let user = User::new("#UNIQUE-USER-IDENTIFIER#");
+let details_with_user = await client.get_all_details(Some(user)).await;
+```
+
+## Cache
+
+The _ConfigCat SDK_ stores the downloaded config data in a local cache to minimize network traffic and enhance client performance.
+If you prefer to use your own cache solution, such as an external or distributed cache in your system,
+you can implement the [`ConfigCache`](https://github.com/configcat/rust-sdk/blob/main/src/cache.rs) trait
+and call the `cache()` method of the `ClientBuilder` with your implementation.
+This allows you to seamlessly integrate ConfigCat with your existing caching infrastructure.
+
+```rust
+struct CustomCache {}
+
+impl ConfigCache for CustomCache {
+    fn read(&self, key: &str) -> Option<String> {
+        // here you have to return with the cached value
+    }
+
+    fn write(&self, key: &str, value: &str) {
+        // here you have to store the new value in the cache
+    }
+}
+```
+
+Then use your custom cache implementation:
+
+```rust 
+let client = Client::builder("#YOUR-SDK-KEY#")
+    // highlight-next-line
+    .cache(Box::new(CustomCache{}))
+    .build()
+    .unwrap();
+```
+
+## HTTP Proxy
+
+The SDK uses the [`reqwest`](https://docs.rs/reqwest) crate for HTTP communication. This crate supports [HTTP proxies](https://docs.rs/reqwest/#proxies) via the `HTTP_PROXY` / `HTTPS_PROXY` environment variables.
+
+## HTTP Timeout
+
+You can set the maximum wait time for a ConfigCat HTTP response.
+
+```csharp
+let client = Client::builder("#YOUR-SDK-KEY#")
+    // highlight-next-line
+    .http_timeout(Duration::from_secs(60));
+    .build()
+    .unwrap();
+```
+
+The default timeout is `30` seconds.
+
+## Logging
```

Comment:
It would be nice to also include some info on how to set the log level (plus the list of possible log levels) and how to implement a custom logger like it's done e.g. [here](https://a6201bf8.docs-xmy.pages.dev/docs/sdk-reference/cpp/#logging)

Reply:
Added a link to a sample log implementation